qiskit/algorithms/gradients/lin_comb_qfi.py

```diff
@@ -0,0 +1,258 @@
+    ):
+        r"""
+        Args:
+            estimator: The estimator used to compute the QFI.
+            phase_fix: Whether to calculate the second term (phase fix) of the QFI, which is
+                Re[(dω⟨<ψ(ω)|)|ψ(ω)><ψ(ω)|(dω|ψ(ω))>]. Default to ``True``.
+            derivative_type: The type of derivative. Can be either ``DerivativeType.REAL``
+                ``DerivativeType.IMAG``, or ``DerivativeType.COMPLEX``. Defaults to
+                ``DerivativeType.REAL``.
+
+                - ``DerivativeType.REAL`` computes :math:`4 \mathrm{Re}[(dω⟨ψ(ω)|)O(θ)|ψ(ω)〉]`.
+                - ``DerivativeType.IMAG`` computes :math:`4 \mathrm{Im}[(dω⟨ψ(ω)|)O(θ)|ψ(ω)〉]`.
+                - ``DerivativeType.COMPLEX`` computes :math:`4 (dω⟨ψ(ω)|)O(θ)|ψ(ω)〉`.
+
+            options: Backend runtime options used for circuit execution. The order of priority is:
+                options in ``run`` method > QFI's default options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+        """
+        super().__init__(options)
+        self._estimator: BaseEstimator = estimator
+        self._phase_fix = phase_fix
+        self._derivative_type = derivative_type
+        self._gradient = LinCombEstimatorGradient(
+            estimator, derivative_type=DerivativeType.COMPLEX, options=options
+        )
+        self._qfi_circuits = {}
+
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter] | None],
+        **options,
+    ) -> QFIResult:
+        """Compute the QFIs on the given circuits."""
+        jobs, result_indices_all, coeffs_all = [], [], []
+        metadata_, gradient_jobs, phase_fixes = [], [], []
+
+        for circuit, parameter_values_, parameters_ in zip(circuits, parameter_values, parameters):
+            # a set of parameters to be differentiated
+            if parameters_ is None:
+                param_set = set(circuit.parameters)
+            else:
+                param_set = set(parameters_)
+
+            meta = {"parameters": [p for p in circuit.parameters if p in param_set]}
+            meta["derivative_type"] = self._derivative_type
+            metadata_.append(meta)
+
+            observable = SparsePauliOp.from_list([("I" * circuit.num_qubits, 1)])
+            # compute the second term (the phase fix) in the QFI
+            if self._phase_fix:
+                gradient_job = self._gradient.run(
+                    circuits=[circuit],
+                    observables=[observable],
+                    parameter_values=[parameter_values_],
+                    parameters=[parameters_],
+                    **options,
+                )
+                gradient_jobs.append(gradient_job)
+            # compute the first term in the QFI
+            qfi_circuits_ = self._qfi_circuits.get(id(circuit))
```

Comment:
Should this use `_circuit_key` instead of the object id?

Reply:
Ah, I didn't know about `_circuit_key`. I will use it. Also, I will make another PR to use `_circuit_key` in other gradient as well.